go/libraries/doltcore/table/untyped/csv/reader.go

```diff
@@ -187,42 +187,57 @@ func (csvr *CSVReader) ReadSqlRow(crx context.Context) (sql.Row, error) {
 		return nil, io.EOF
 	}
 
-	colVals, err := csvr.csvReadRecords(nil)
+	rowVals, err := csvr.csvReadRecords(nil)
 
 	if err == io.EOF {
 		csvr.isDone = true
 		return nil, io.EOF
 	}
 
 	schSize := csvr.sch.GetAllCols().Size()
-	if len(colVals) != schSize {
+	if len(rowVals) != schSize {
 		var out strings.Builder
-		for _, cv := range colVals {
+		for _, cv := range rowVals {
 			if cv != nil {
 				out.WriteString(*cv)
 			}
 			out.WriteRune(',')
 		}
-		return nil, table.NewBadRow(nil,
-			fmt.Sprintf("csv reader's schema expects %d fields, but line only has %d values.", schSize, len(colVals)),
-			fmt.Sprintf("line: '%s'", out.String()),
+
+		badMpStr, unusedRowValues := interpretRowSizeError(csvr.sch, rowVals)
+
+		args := []string{
+			fmt.Sprintf("CSV reader expected %d values, but saw %d.", schSize, len(rowVals)),
+			fmt.Sprintf("The row was interpreted to be: '%s'", badMpStr),
+		}
```

Comment:
How hard would it be to include the line number of the bad row? That seems like the gold standard for reporting an error in a data file and what customers would likely expect.

Reply:
I like this feature idea. I'll put it in a separate pr. It's something that works across file types